storage/repository/errors.go

```diff
@@ -0,0 +1,32 @@
+package repository
+
+// Declare conformance with Error interface
+var _ error = OperationNotSupportedError
+
+type operationNotSupportedError string
+
+// OperationNotSupportedError may be thrown by certain functions in the storage
+// or repository packages which operate on URIs if an operation is attempted
+// that is not supported for the scheme relevant to the URI, normally because
+// the underlying repository has either not implemented the relevant function,
+// or has explicitly returned this error.
+const OperationNotSupportedError operationNotSupportedError = operationNotSupportedError("Operation not supported for this URI.")
```

Comment:
Would `var ErrOperationNotSupported = errors.New("Operation not supported for this URI.")` be simpler?

Reply:
I cargo-culted this from `URIRootError`, which I assumed was right. I don't remember why we did it differently there.

Reply:
So that it is a const nnot a var

Reply:
On the call we agreed that wasn't an issue.

Reply:
Also I prefer a common prefix like `TextAlignLeading, TextAlignCenter, TextAlignTrailing`, so `ErrOperationNotSupported, ErrURIRoot, ErrABC, ErrXYZ, ...`

Reply:
I have no strong feeling - abstain from this debate, but OK with whatever solution folks like.

Reply:
Any further thoughts on this from folks?

Reply:
Seems I'm the only one with an opinion on this.
As this would be the first time Fyne declares and exports errors for devs to compare, there is no internal precedence to follow, and in such a case I would recommend following the precedence set by Go's standard libraries.

Reply:
To my knowledge, `OperationNotSupportedError` follows the conventions of the Go stdlib. Is there another specific name you had in mind?

Reply:
`OperationNotSupported` is ok, I was referring to `var XYZ = errors.New("XYZ")`

Reply:
This is not the first error to be exposed - `const storage.URIRootError` was in 1.4 and as far as I can see that is what @charlesdaniels is matching.

If we should move to var errors I don't know, the error spec is a moving target at the moment.

Reply:
After a chat in Slack and Jitsu we agreed to follow a spec and found these;
- https://github.com/golang/go/wiki/Errors
- https://blog.golang.org/go1.13-errors

Leading us to;
```go
var (
    ErrOperationNotSupported = errors.New("Operation not supported for this URI.")
    ErrURIRoot               = errors.New("Cannot take the parent of the root element in a URI")
)
```

Reply:
c0dc44a2d1c73ea6fb37a30063b367dc70bc2707